Kubernetes/helmcharts/azure-policy-addon-arc-connected-clusters/templates/remove-addon.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: pre-delete-admin
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pre-delete-agent
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+rules:
+- apiGroups: ["apps"]
+  resources: ["deployments"]
+  verbs: ["delete"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: pre-delete-agent
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "20"
+subjects:
+- kind: ServiceAccount
+  name: pre-delete-admin
+  namespace: kube-system
+roleRef:
+  kind: Role
+  name: pre-delete-agent
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
```

Comment:
I see this pre-delete hook job is removing the azure-policy pod which has been deployed through helm chart? Would there be any race conditions in below scenarios...
1. After pre-delete hook execution, helm tries to uninstall the pod but the pod has already been deleted. Would this cause any error?
2. Azure-Policy pod got deleted but helm hasn't yet finished uninstall. Before chart is uninstalled, would there be a scenario where Kubernetes master server sees pod missing and tries to create a new pod?